### PR TITLE
Benchmarking

### PR DESCRIPTION
**Pull request by d0nut-grapl**
### Which issue does this PR correspond to?

Benchmark issue: 416

### What changes does this PR make to Grapl? Why?

Adds basic benchmarks to our generators (aside from generic generators).

### How were these changes tested?

Tested locally and verified benchmarks work.

### CODE REVIEW

File: src/rust/generators/sysmon-subgraph-generator/benches/generator_bench.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+use grapl_test::cache::EmptyCache;
+use sqs_executor::{event_decoder::PayloadDecoder,
+                   event_handler::CompletedEvents};
+use sysmon_subgraph_generator_lib::{generator::SysmonSubgraphGenerator,
+                                    metrics::SysmonSubgraphGeneratorMetrics,
+                                    serialization::SysmonDecoder};
+use test::Bencher;
+use tokio::runtime::Runtime;
+
+const SYSMON_SAMPLE_DATA_FILE: &'static str = "../../../../etc/sample_data/events6.xml";
```
Comment (inickles-grapl) [inline src/rust/generators/sysmon-subgraph-generator/benches/generator_bench.rs:17]: I know we're mono repo, but in the future I'd like to see this kind of thing be more self contained. In this case, that'd be moving to events6.xml to the sysmon generator directory. It's not like that file is useful without that generator.

Not to say I wanna require that for this PR, but  if you agree I'll bring it up for discussion.

File: src/rust/rust-toolchain [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.51.0"
+channel = "nightly"
```
Comment (inickles-grapl) [inline src/rust/rust-toolchain:2]: We don't wanna do this.

File: src/rust/generators/sysmon-subgraph-generator/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#![type_length_limit = "1334469"]
```
Comment (inickles-grapl) [inline src/rust/generators/sysmon-subgraph-generator/src/lib.rs:1]: Are these needed anymore? As I recall they were some workaround to a weird issue in Rust a while back.

**Review by inickles-grapl — CHANGES_REQUESTED** [review]
Considering we should use `sqs-executor::cache::NopCache` instead of the new `EmptyCache`, seems we don't need the new grapl-test crate.

Also, we shouldn't change the rust toolchain from 1.51 to nightly.

File: src/rust/generators/sysmon-subgraph-generator/benches/generator_bench.rs (hunk 1 above)
Comment (colin-grapl) [reply src/rust/generators/sysmon-subgraph-generator/benches/generator_bench.rs:17]: Yeah, I think we should use a dedicated benchmark file - events6.xml could move, or change in the future.

**Review by colin-grapl — CHANGES_REQUESTED** [review]

File: src/rust/grapl-test/src/cache.rs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+use sqs_executor::cache::{Cache,
+                          Cacheable};
+
+use crate::error::GenericError;
+
+#[derive(Clone)]
+pub struct EmptyCache {}
+
+#[async_trait]
+impl Cache for EmptyCache {
```
[the remark this answers is not in the document]
Comment (d0nut-grapl) [reply src/rust/grapl-test/src/cache.rs:11]: Oh I didn't know about that! Thanks!

File: src/rust/rust-toolchain (hunk 2 above)
Comment (d0nut-grapl) [reply src/rust/rust-toolchain:2]: Ahh yea I forgot to revert this. I had to fix a bug that was caused by this file actually

File: src/rust/generators/sysmon-subgraph-generator/benches/generator_bench.rs (hunk 1 above)
Comment (d0nut-grapl) [reply src/rust/generators/sysmon-subgraph-generator/benches/generator_bench.rs:17]: Will move a copy of the test data into the repos specifically! Sorry; got lazy during hack week :)

The code limits the number of parsed events that each generator is processing so the results should be relatively consistent and comparable.

**Review by inickles-grapl — APPROVED** [review]

**Review by colin-grapl — APPROVED** [review]

**Pull request merged**